src/command-helpers/network.js

```diff
@@ -0,0 +1,135 @@
+        // so we have to update the network of all templates too.
+        if(subgraph.templates) {
+          subgraph.templates = updateTemplates(network, subgraph.templates)
+        }
+
+        let unsusedSources = networkSources.filter(x => !subgraphSources.includes(x))
+
+        unsusedSources.forEach(source => {
+          step(spinner, `dataSource '${source}' from '${networksFile}' not found in ${manifest}`)
+        })
+
+        let yaml_doc = new yaml.Document()
+        yaml_doc.contents = subgraph
+        return yaml_doc.toString()
+      })
+  })
+
+const initNetworksConfig = async(toolbox, directory) =>
+  await withSpinner(
+    `Initialize networks config`,
+    `Failed to initialize networks config`,
+    `Warnings while initializing networks config`,
+    async spinner => {
+      let subgraphStr = await toolbox.filesystem.read(path.join(directory, 'subgraph.yaml'))
+      let subgraph = yaml.parse(subgraphStr)
+
+      let networks = {}
+
+      subgraph.dataSources.forEach(source => {
+        let sourceConfig = {[source.name]: {}}
+
+        if (source.source.address) sourceConfig[source.name]["address"] = source.source.address
+        if (source.source.startBlock) sourceConfig[source.name]["startBlock"] = source.source.startBlock
+
+        networks[source.network] = sourceConfig
+      })
+
+      await toolbox.filesystem.write(`${directory}/networks.json`, networks)
+
+      return true
+    },
+  )
+
+// Iterates through each dataSource and updates the network configuration
+// Will update only if there are changes to the configuration
+function updateSources(toolbox, spinner, network, sources, networkSources, networkConfig) {
+  sources.forEach(source => {
+      if (!networkSources.includes(source.name)) {
+        throw new Error(`'${source.name}' was not found in the '${network}' configuration, please update!`)
+      }
+
+      if (hasChanges(network, networkConfig[source.name], source)) {
+        step(spinner, `Update '${source.name}' network configuration`)
+        source.network = network
+        source.source = { abi: source.source.abi }
+        Object.assign(source.source, networkConfig[source.name])
+      } else {
+        step(spinner, `Skip '${source.name}': No changes to network configuration`)
+      }
+  })
+
+  return sources
+}
+
+// Iterates through each template and updates the network
+function updateTemplates(network, templates) {
+  templates.forEach(template => {
+    if (template.network !== network) template.network = network
+  })
+
+  return templates
+}
+
+// Checks if any network attribute has been changed
+function hasChanges(network, networkConfig, dataSource) {
+  let networkChanged = dataSource.network !== network
+
+  // Return directly if the network is different
+  if (networkChanged) return networkChanged
+
+  let addressChanged = (network === "near")
+    ? networkConfig.account !== dataSource.source.account
+    : networkConfig.address !== dataSource.source.address
```

Comment:
I think there should be no chain/protocol specific logic outside of the `protocols` module/folder, and this is already there _I think_. I didn't think it through yet, what's needed to get an instance of the `Protocol` class here, but that would be preferable 🙂

Reply:
I looked at it, but I wasn't sure how I could utilise it in this case

Reply:
So what I mean is that on the commands `build` and `init` you should pass the `protocol` instance and call the [`getContract` function](https://github.com/graphprotocol/graph-cli/blob/main/src/protocols/index.js#L191), and then the [`identifierName` one](https://github.com/graphprotocol/graph-cli/blob/main/src/protocols/ethereum/contract.js#L3), so that we don't spread `protocol` logic outside of that module.

Reply:
Here, for the `build` command I'm checking the name of the network passed with the `--network` option.
Not sure the protocol instance will be of help in this case, but I just realised I'm not handling `near` contracts in the `init` command, and I can use it for that there.